### PR TITLE
Add support for virtual networks

### LINKED ISSUE #1602: Feature: add cloudflare_tunnel_virtual_network resource
### Current Terraform and Cloudflare provider version

Terraform v1.1.4
+ provider registry.terraform.io/cloudflare/cloudflare v3.14.0

### Description

* Add the Terraform resource for Tunnel Virtual Networks
https://api.cloudflare.com/#tunnel-virtual-network-properties

### Use cases

It would be really handy to have an option to manage Tunnel Virtual Networks and assignment of Tunnels through Terraform instead of CLI/API. 

### Potential Terraform configuration

```hcl
resource "cloudflare_tunnel_virtual_network" "example" {
  name = "us-east-1-vpc"
  is_default_network = true
  comment = Staging VPC for data science
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by vavsab**
Add missing vnet resource

Related issues: #1602 cloudflare/cloudflare-go#913

Closes #1602

### CODE REVIEW

**Comment by jacobbednarz** [comment]
as this is a new resource, can you please swap over to using the autogenerated docs by following the example in #1677? it should only require:

- adding the descriptions to the schema fields
- adding either example usage
- adding the import example
- running `make docs`

I'll review the functionality in the AM.

**Comment by vavsab** [comment]
> as this is a new resource, can you please swap over to using the autogenerated docs by following the example in #1677? it should only require:
> 
> * adding the descriptions to the schema fields
> * adding either example usage
> * adding the import example
> * running `make docs`
> 
> I'll review the functionality in the AM.

@jacobbednarz created examples and generated the docs

File: examples/resources/cloudflare_tunnel_virtual_network/import.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+terraform import cloudflare_tunnel_virtual_network c4a7362d577a6c3019a474fd6f485821/3c8ff8af-b487-45bd-89e3-4c85a1532600
```
Comment (jacobbednarz) [inline examples/resources/cloudflare_tunnel_virtual_network/import.sh:1]:
```suggestion
$ terraform import cloudflare_tunnel_virtual_network c4a7362d577a6c3019a474fd6f485821/3c8ff8af-b487-45bd-89e3-4c85a1532600
```

**Comment by jacobbednarz** [comment]
looks like the update comment test is failing

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareTunnelVirtualNetwork_" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareTunnelVirtualNetwork_Exists
--- PASS: TestAccCloudflareTunnelVirtualNetwork_Exists (14.70s)
=== RUN   TestAccCloudflareTunnelVirtualNetwork_UpdateComment
    resource_cloudflare_tunnel_virtual_network_test.go:118: Step 2/2 error: Error running apply: exit status 1

        Error: error updating Tunnel Virtual Network "4760e5b2-c737-4b8d-98f9-9ce33b581d7c": error unmarshalling the JSON response error body: unexpected end of JSON input

          with cloudflare_tunnel_virtual_network.xuysfsklky,
          on terraform_plugin_test.tf line 2, in resource "cloudflare_tunnel_virtual_network" "xuysfsklky":
           2: resource "cloudflare_tunnel_virtual_network" "xuysfsklky" {

--- FAIL: TestAccCloudflareTunnelVirtualNetwork_UpdateComment (18.02s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/provider	33.292s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
FAIL
make: *** [testacc] Error 1
```

from the HTTP traffic, it looks like the virtual network ID is missing in the PATCH request.

File: internal/provider/resource_cloudflare_tunnel_virtual_network.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	return resourceCloudflareTunnelVirtualNetworkRead(ctx, d, meta)
+}
+
+func resourceCloudflareTunnelVirtualNetworkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+
+	resource := cloudflare.TunnelVirtualNetworkUpdateParams{
+		AccountID:        d.Get("account_id").(string),
+		Name:             d.Get("name").(string),
+		IsDefaultNetwork: cloudflare.BoolPtr(d.Get("is_default_network").(bool)),
```
Comment (jacobbednarz) [inline internal/provider/resource_cloudflare_tunnel_virtual_network.go:94]: need to include `VnetID` in the parameters to get the tests passing

```suggestion
		IsDefaultNetwork: cloudflare.BoolPtr(d.Get("is_default_network").(bool)),
		VnetID:           d.Id(),
```

**Comment by jacobbednarz** [comment]
couple of small things and we can get this one merged in

**Comment by vavsab** [comment]
All green now
image

**Comment by jacobbednarz** [comment]
acceptance tests all passing here

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareTunnelVirtualNetwork_" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareTunnelVirtualNetwork_Exists
--- PASS: TestAccCloudflareTunnelVirtualNetwork_Exists (13.33s)
=== RUN   TestAccCloudflareTunnelVirtualNetwork_UpdateComment
--- PASS: TestAccCloudflareTunnelVirtualNetwork_UpdateComment (24.03s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/provider	37.941s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
```

thanks for getting this over the line!

**Pull request merged**